### PR TITLE
GP Internal Core API v1.1 : review property access functions

### PR DESCRIPTION
**Pull request by cedric-chaumont-st-dev**
update size_t to uint32_t type, panic

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
I'm not sure that replacing all occurrences of `size_t` with `uint32_t` is the best solution. I would prefer if we tried to limit to changing to only the argument parameters of the exposed functions and as few local variables and functions as long as it's convenient.

`size_t` is the preferred type to used when dealing with sizes of buffers etc, replacing it with `uint32_t` works great on a 32-bit platform, but on a 64-bit platform it's perhaps not so good.

**Comment by pascal-brand-st-dev** [comment]
+1 on Jens's opinion on size_t vs uint32_t

**Comment by cedric-chaumont-st-dev** [comment]
+1. Updated

File: lib/libutee/tee_api_property.c [hunk 1]
```diff
@@ -327,15 +329,27 @@ TEE_Result TEE_GetPropertyAsString(TEE_PropSetHandle propsetOrEnumerator,
...
+		goto err;
+	}
+
+	goto out;
 
+err:
+	if (res == TEE_ERROR_ITEM_NOT_FOUND ||
+	    res == TEE_ERROR_SHORT_BUFFER)
+		return res;
+	else
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: There's a few checkpatch warnings on else after return.

File: lib/libutee/tee_api_property.c [hunk 2]
```diff
@@ -255,7 +255,7 @@ static TEE_Result propget_get_property(TEE_PropSetHandle h, char *name,
 		if (idx < eps_len)
 			return propget_get_ext_prop(eps + idx, pv);
 
-		return TEE_ERROR_BAD_PARAMETERS;
+		return TEE_ERROR_ITEM_NOT_FOUND;
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: Why this change?

**Comment by cedric-chaumont-st-dev** [comment]
Update

File: lib/libutee/tee_api_property.c [hunk 3]
```diff
@@ -387,27 +426,42 @@ TEE_Result TEE_GetPropertyAsBinaryBlock(TEE_PropSetHandle propsetOrEnumerator,
...
+	if (pv.type != USER_TA_PROP_TYPE_BINARY_BLOCK) {
+		res = TEE_ERROR_BAD_FORMAT;
+		goto err;
+	}
 
 	val = pv.u.str_val;
 	val_len = strlen(val);
 	size = *valueBufferLen;
-	if (!base64_dec(val, val_len, valueBuffer, &size))
+	if (!base64_dec(val, val_len, valueBuffer, &size)) {
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: base64_dec() expects `size_t *` in 4th parameter, now you're supplying `uint32_t *` instead.

File: lib/libutee/tee_api_property.c [hunk 4]
```diff
@@ -265,17 +265,19 @@ TEE_Result TEE_GetPropertyAsString(TEE_PropSetHandle propsetOrEnumerator,
 {
 	TEE_Result res;
 	struct prop_value pv;
-	size_t l;
-	size_t bufferlen;
+	uint32_t l;
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: `l` and `bufferlen` doesn't need to be `uint32_t`, or am I missing something?

File: lib/libutee/tee_api_property.c [hunk 5]
```diff
@@ -502,33 +595,47 @@ TEE_Result TEE_GetPropertyName(TEE_PropSetHandle enumerator,
 	size_t ps_len;
 	const struct user_ta_property *eps;
 	size_t eps_len;
-	size_t l;
+	uint32_t l;
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: `l` and `bufferlen` doesn't have to be `uint32_t`

Comment (cedric-chaumont-st-dev) [inline lib/libutee/tee_api_property.c]: OK

**Comment by cedric-chaumont-st-dev** [comment]
+1

**Comment by cedric-chaumont-st-dev** [comment]
+1

**Comment by cedric-chaumont-st-dev** [comment]
Change TEE_ERROR_ITEM_NOT_FOUND to TEE_ERROR_GENERIC in propset_get() function breaks GP compliance test  a7-83-30. I would prefer not to modify the TEE_Internal_API.xml.

**Comment by pascal-brand-st-dev** [comment]
Take care of the final commit message as the uint32_t change is no more there.

    Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Comment by pascal-brand-st-dev** [comment]
Tested-by: Pascal Brand <pascal.brand@linaro.org> (STM platform)
